test/app-tests/wrapper-test.tsx

```diff
@@ -0,0 +1,90 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import SdkConfig from "matrix-react-sdk/src/SdkConfig";
+import PlatformPeg from "matrix-react-sdk/src/PlatformPeg";
+import fetchMock from "fetch-mock-jest";
+import { render, RenderResult, screen, waitFor } from "@testing-library/react";
+import { ModuleRunner } from "matrix-react-sdk/src/modules/ModuleRunner";
+import { WrapperLifecycle, WrapperOpts } from "@matrix-org/react-sdk-module-api/lib/lifecycles/WrapperLifecycle";
+
+import WebPlatform from "../../src/vector/platform/WebPlatform";
+import { loadApp } from "../../src/vector/app";
+
+fetchMock.config.overwriteRoutes = true;
+
+describe("Wrapper", () => {
+    beforeEach(async () => {
+        SdkConfig.reset();
+        PlatformPeg.set(new WebPlatform());
+        fetchMock.get("https://matrix-client.matrix.org/_matrix/client/versions", {
+            unstable_features: {},
+            versions: ["v1.1"],
+        });
+        fetchMock.get("https://matrix.org/.well-known/matrix/client", {
+            "m.homeserver": {
+                base_url: "https://matrix-client.matrix.org",
+            },
+        });
+        fetchMock.get("/version", "1.10.13");
+    });
+
+    it("wrap a matrix client with header and footer", async () => {
+        SdkConfig.put({
+            default_server_config: {
+                "m.homeserver": {
+                    base_url: "https://matrix-client.matrix.org",
+                },
+            },
+        });
+
+        jest.spyOn(ModuleRunner.instance, "invoke").mockImplementation((lifecycleEvent, opts) => {
+            if (lifecycleEvent === WrapperLifecycle.Wrapper) {
+                (opts as WrapperOpts).Wrapper = ({ children }) => {
+                    return (
+                        <>
+                            <div>Header</div>
+                            {children}
+                            <div>Footer</div>
+                        </>
+                    );
+                };
+            }
+        });
+
+        const matrixChat: RenderResult = render(await loadApp({}));
+
+        // at this point, we're trying to do a guest registration;
+        // we expect a spinner
+        await assertAtLoadingSpinner();
+
+        await awaitWelcomeComponent(matrixChat);
+
+        // Are not semantic elements because Element has a footer already.
+        screen.getByText(/Header/i);
+        screen.getByText(/Footer/i);
+    });
+});
+
+async function assertAtLoadingSpinner(): Promise<void> {
+    await screen.findByRole("progressbar");
+}
+
+async function awaitWelcomeComponent(matrixChat?: RenderResult): Promise<void> {
```

Comment:
Maybe `waitFor` rather than `await`? Otherwise it reads `await await...`

Reply:
it was copied from 'loading-test.tsx' to follow the same pattern, no problem to rename it, but it will be inconsistent. I can also move it to separate file and reuse between these two tests. What do you think?

Reply:
Sounds good, let's do that 👍

Reply:
renamed and moved to separate file.